src/main/java/com/iota/iri/service/tipselection/impl/EntryPointSelectorCumulativeWeightThreshold.java

```diff
@@ -37,16 +37,41 @@ public Hash getEntryPoint() throws Exception {
         return backtrack(tipsViewModel.getRandomSolidTipHash(), threshold);
     }
 
+    /**
+     * Backtracks from tip until it reaches a transaction with a cumulative weight
+     * of at least threshold.
+     * 
+     * Uses an exponential search to avoid running the expensive CW computations
+     * more than necessary.
+     */
     public Hash backtrack(Hash tip, int threshold) throws Exception {
         Hash currentHash = tip;
-        Integer currentWeight = 0;
+        Integer currentWeight = 1;
+        int stepSize = 1;
 
         // Backtrack as long as the genesis hasn't been reached and the tresholed has not been crossed
-        while (currentWeight < threshold && !Objects.equals(currentHash, Hash.NULL_HASH)) {
-            currentHash = TransactionViewModel.fromHash(tangle, currentHash).getTrunkTransactionHash();
+        while (currentWeight < threshold && !isGenesis(currentHash)) {
```

Comment:
As `currentWeight` is an Integer this could in theory throw a `NullPointerException`. I think `calculateSingle` and thus also `currentWeight` should rather be `int` to clarify that `null` is not possible here.